Charmcraft Go tutorial: Add structure to sections (#2339)
canonical/charmcraft: docs/tutorial/kubernetes-charm-go.rst

    setting a DNS record.

The development cycle
---------------------

> So far, we have worked though the entire cycle, from creating an app to deploying it.
> But now – as in every real-world case – we will go through the experience of iterating to
> develop the app, and deploy each iteration.

Reviewer: ```suggestion
So far, we've worked though the entire cycle, from creating an app to deploying
it. But now -- as in every real-world case -- we'll go through the experience of
iteratively developing the app, and deploying each iteration.
```

The dashes are quite minor, but it's best if we let Sphinx handle them in the source.
